### ISSUE TITLE
Credentials enhancement with ssh key use

### ISSUE THREAD

**Issue by AlanKane640** [nautobot/nautobot-app-device-onboarding#189]
### Environment
* Nautobot version:  *
* nautobot-device-onboarding version:  *

### Proposed Functionality
Connecting to devices using ssh key.

### Use Case
The user should have the choice to use ssh key to connect to devices either by using Nautobot secret key or by specifying NAPALM options in a new form field (JSON format).

Thanks.

**Comment by jeffkala** [comment]
Should be fixed in #275, going to test locally and come up with a guide to get it working

**Comment by jeffkala** [comment]
Validated #275 did allow for this.

You'd just have to set the options in the PLUGIN_CONFIG.  Example of a working solution.

```python
PLUGINS = ["nautobot_device_onboarding", "nautobot_ssot", "nautobot_plugin_nornir"]

PLUGINS_CONFIG = {
    "nautobot_device_onboarding": {},
    "nautobot_ssot": {
        "hide_example_jobs": is_truthy(os.getenv("NAUTOBOT_SSOT_HIDE_EXAMPLE_JOBS")),
    },
    "nautobot_plugin_nornir": {
        "nornir_settings": {
            "credentials": "nautobot_plugin_nornir.plugins.credentials.nautobot_secrets.CredentialsNautobotSecrets",
            "runner": {
                "plugin": "threaded",
                "options": {
                    "num_workers": 20,
                },
            },
        },
        "connection_options": {
            "netmiko": {
                "extras": {
                    "use_keys": True,
                    "key_file": "/root/.ssh/id_rsa.pub",
                    "disabled_algorithms": {"pubkeys": ["rsa-sha2-256", "rsa-sha2-512"]},
                },
            },
        },
    },
}
```

I did have to make a <partially bogus> secrets-group in Nautobot which still had all the elements (username and password) where the username has to be accurate, but the password can just be anything as its ignored by the backend processing.
